src/ProjectOrigin.Vault/CommandHandlers/CheckForWithdrawnCertificatesCommandHandler.cs

```diff
@@ -0,0 +1,108 @@
+        ILogger<CheckForWithdrawnCertificatesCommandHandler> logger)
+    {
+        _unitOfWork = unitOfWork;
+        _httpClientFactory = httpClientFactory;
+        _logger = logger;
+        _networkOptions = networkOptions.Value;
+    }
+
+    public async Task Consume(ConsumeContext<CheckForWithdrawnCertificatesCommand> context)
+    {
+        var stamps = _networkOptions.Stamps;
+
+        foreach (var stamp in stamps)
+        {
+            var cursors = await _unitOfWork.WithdrawnCursorRepository.GetWithdrawnCursors();
+            var matchingCursor = cursors.FirstOrDefault(x => x.StampName == stamp.Key) ?? new WithdrawnCursor
+            {
+                StampName = stamp.Key,
+                SyncPosition = 0,
+                LastSyncDate = DateTimeOffset.UtcNow
+            };
+
+            var client = _httpClientFactory.CreateClient();
+            var response = (await client.GetFromJsonAsync<ResultList<WithdrawnCertificateDto, PageInfo>>(stamp.Value.Url + $"/v1/certificates/withdrawn?lastWithdrawnId={matchingCursor.SyncPosition}"))!;
+
+            if (!response.Result.Any())
+            {
+                _logger.LogInformation("No withdrawn certificates found for {StampName}", stamp.Key);
+                continue;
+            }
+
+            _logger.LogInformation("Found {Count} withdrawn certificates for {StampName}", response.Result.Count(), stamp.Key);
+            foreach (var withdrawnCertificate in response.Result)
+            {
+                await _unitOfWork.CertificateRepository.WithdrawCertificate(withdrawnCertificate.RegistryName, withdrawnCertificate.CertificateId);
+                var claimedSlices = await _unitOfWork.CertificateRepository.GetClaimedSlicesOfCertificate(withdrawnCertificate.RegistryName, withdrawnCertificate.CertificateId);
+                foreach (var claimedSlice in claimedSlices)
+                {
+                    //Unclaim (which is next task)
+                }
+            }
+
+            matchingCursor.SyncPosition = response.Result.Max(x => x.Id);
+            matchingCursor.LastSyncDate = DateTimeOffset.UtcNow;
+            await UpdateWithdrawnCursor(matchingCursor);
+            _unitOfWork.Commit();
+        }
+    }
+
+    private async Task UpdateWithdrawnCursor(WithdrawnCursor updatedCursor)
+    {
+        var cursors = await _unitOfWork.WithdrawnCursorRepository.GetWithdrawnCursors();
+        var matchingCursor = cursors.FirstOrDefault(x => x.StampName == updatedCursor.StampName);
+        if (matchingCursor == null)
+            await _unitOfWork.WithdrawnCursorRepository.InsertWithdrawnCursor(updatedCursor);
+        else
+            await _unitOfWork.WithdrawnCursorRepository.UpdateWithdrawnCursor(updatedCursor);
+    }
+}
+public record ResultList<T, TPageInfo>()
+{
+    public required IEnumerable<T> Result { get; init; }
+    public required TPageInfo Metadata { get; init; }
+}
+
+public record PageInfo()
```

Comment:
Shouldn't this be part of shared package for PO. Remember we had something, for stuff that was used by several projects.

Reply:
Then we couple it to Services.REST.v1.SharedModels but sure - they do not change atm